### DISCUSSION TITLE
トラックボールICからのマウス移動量の入力でチャタリングが発生する

### DISCUSSION THREAD

**Discussion by moegikaede** [Yowkees/keyball#289] [Q&A]
タイトルの通り、トラックボールICからのマウス移動量の取得時にチャタリング（127, -127の入力が突然入る）が発生している状態となっています。
ProMicroからトラックボール用ピンまで VCC、GND、CLK、MIS0、MOS1、MCSは問題なく通電できていることは確認済みです。

チャタリングの原因となるHW的な要素を何か知っていればご教授いただければと思います。
また、qmkのコンパイルでチャタリングに対応したファームウェアを試してチャタリング対策はできたのですが、このファームウェアでは Remapが使えなくなってしまっているので Remap入りのファームウェアの作成方法があればこちらもお教えいただきたいです。

**Comment by moegikaede** [comment]
他の方の不具合内容を確認していたところ #282、#234、#238と同様の現象のようです。
白銀ラボの問い合わせで追って連絡させていただきます。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。
可能性の１つとして、L字7PコンスルーのU字型のピンが変形して、接触不良となっている可能性があります（過去に経験があります）。
具体的には、コンスルーのU字型のピン（半田付けしないピン）が横から見たときにつぶれて細くなっている場合があり、ピンセット等で幅を広げてあげると安定して導通するようになります。
一度トラックボール基板を外して確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。

**Comment by moegikaede** [comment]
ピンセットでL字7PコンスルーのU字型のピンをピンセットで広げた後、Remap対応のファームウェアに書き換えてもチャタリングが発生しなくなりました。
他の方（#282、#234、#238）も同様の現象でしたので同じ手法で改善するかもしれません。

サポートが早く、的確で非常に助かりました。
使い慣れて手放せなくなったら自宅用にまた買わせていただくと思います。
本当にありがとうございました。

**Comment by Yowkees** [comment]
無事に改善できてよかったです！ぜひご検討いただければ幸いです。
